### DISCUSSION TITLE
Add x-model to <input> programmatically

### DISCUSSION THREAD

**Discussion by AndiLeni** [alpinejs/alpine#3897] [1. Help] [answered]
Hello,

it is somehow possible to programmatically add the x-model directive to an input field?
In my usecase, I have <input>s, <select>s, ..., fields but I can not add "x-model" directly to the html element because I have no access to the template directly which outputs these fields.
I'd like to make a field reactive via javascript.

Thanks and kind regards

**Comment by ekwoka** [comment, marked as answer]
You can just go through them and set the attribute, or use `Alpine.bind` to add them without actually setting the attribute

**Reply by AndiLeni** [reply]
Thanks!

**Reply by FLWFTW** [reply]
@ekwoka Could I get you to expand on that? if I had `const x = document.getElementsByClassName('custom')` and I looped through all elements in the `x` array, how would I set the x-model attribute on them?

I appreciate any guidance you can give!

Thanks!

**Reply by ekwoka** [reply]
anything like

```js
document.querySelectorAll('.custom').forEach((el,i) => el.setAttribute('x-model',`items[${i}]`))
```

Or

```js
document.querySelectorAll('.custom').forEach((el,i) => Alpine.bind(el, { 'x-model':`items[${i}]` }))
```

**Reply by FLWFTW** [reply]
Oh wow that was easy, thank you!!
